### DISCUSSION TITLE
Add ability to disable particular Expo Doctor tests

### DISCUSSION THREAD

**Discussion by floatplane** [expo/expo#25698] [Proposals and Ideas]
# Background

We are currently running Expo Doctor in CI for all pull requests in our repo (https://github.com/NWACus/avy). This is generally pretty comforting, and in one case [actually caught a problem](https://github.com/NWACus/avy/actions/runs/6647613290/job/18063317132?pr=489) where the wrong size icon was used. 

But it's *way* more common for Expo Doctor to fail the `Check that packages match versions required by installed Expo SDK` check, and I really don't want those failures in CI:

- This check is not related to the code in the PR. It's purely a function of calendar time and things happening async to the PR. The same PR submitted 24 hours earlier might not fail.
- We want to be careful about incrementing package versions, and not just do it to fix a PR build failure. Updating new package versions often require a new binary update, as opposed to an OTA bundle update. 

Anecdote: I caused a crashlooping problem in early testing when I just applied the recommended updates and pushed them out OTA to our testers. The app binary downloaded an OTA update that it couldn't run, and just kept restarting. Even after unpublishing the OTA update, any user who happened to pull the update had to do a full app delete and reinstall to get things working again - not great! I would never want this to happen in release. (I see a [comment saying that Expo Doctor should not recommend an upgrade that is incompatible](https://github.com/expo/expo/discussions/18568#discussioncomment-3376819), but [my experience](https://github.com/NWACus/avy/issues/556) indicates that this is not true).

# Proposal

## Easy solution
Add a `--skip-package-version-check` argument to Expo Doctor, allowing this particular check to be skipped

## Slightly more involved solution
Add a more general `--skip` argument that allows skipping any check by name. More flexible but requires a few more changes to Expo Doctor:
  - Add a proper argument parsing library
  - Need short names for all checks
  - Update help text to display the short names so users can figure out how to disable the checks they want to skip

---

Let me know if you'd take a PR with either of the solutions above! Otherwise I'll probably end up writing a wrapper around Expo Doctor that parses the output and ignores the failures I'm not interested in.

**Comment by keith-kurak** [comment]
Hi! We do intend to introduce more general options for skipping certain checks, but want to get through refactoring the organization of those checks first, which would change all the flags. Therefore, instead of enshrining a commandline option that could change soon, we thought that maybe it could be an environment variable. It can work the same, since you can inline it, e.g., `EXPO_DOCTOR_SKIP_DEPENDENCY_VERSION_CHECK=1 npx expo-doctor@latest`. 

We'd happily accept a PR on this and I'm happy to work with you to get it merged. Heads up that we have just moved Doctor to the main expo/expo repo (under **packages**). DM me on discord (I'm `@keith`) and we can discuss further!

**Reply by floatplane** [reply]
Cool, that idea works well for me! I'll work on a PR later today/this weekend and ping you on Discord with any issues. Thanks!
